lib/rules/no-param-reassign.js

```diff
@@ -22,7 +22,13 @@ module.exports = {
             {
                 type: "object",
                 properties: {
-                    props: { type: "boolean" }
+                    props: { type: "boolean" },
+                    ignorePropertyAssignmentsFor: {
```

Comment:
Since `ignorePropertyAssignmentsFor` only works for `props: true`, I **think** you could use `oneOf` and `enum` to restrict the schema to such case.